### DISCUSSION TITLE
Need code to Create cross account alarms - i.e create alarms in monitoring account  from the metrics which is on another account.

### DISCUSSION THREAD

**Discussion by preeti3012** [gruntwork-io/knowledge-base#139] [Help] [answered]
Need code to Create cross account alarms - i.e create alarms in monitoring account  from the metrics which is on another account.

Using Terrgarunt  and terraform in code.

r:terraform-aws-monitoring

**Comment by gruntwork-discussions** [comment]
You must label this issue with at least one of the following labels before it can be answered:

* [r:terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)
* [r:terraform-aws-vpc](https://github.com/gruntwork-io/terraform-aws-vpc)
* [r:terraform-aws-architecture-catalog](https://github.com/gruntwork-io/terraform-aws-architecture-catalog)
* [r:terraform-aws-asg](https://github.com/gruntwork-io/terraform-aws-asg)
* [r:terraform-aws-beanstalk](https://github.com/gruntwork-io/terraform-aws-beanstalk)
* [r:terraform-aws-cache](https://github.com/gruntwork-io/terraform-aws-cache)
* [r:terraform-aws-ci](https://github.com/gruntwork-io/terraform-aws-ci)
* [r:terraform-aws-ci-pipeline-example](https://github.com/gruntwork-io/terraform-aws-ci-pipeline-example)
* [r:terraform-aws-cis-service-catalog](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog)
* [r:terraform-aws-data-storage](https://github.com/gruntwork-io/terraform-aws-data-storage)
* [r:terraform-aws-ecs](https://github.com/gruntwork-io/terraform-aws-ecs)
* [r:terraform-aws-eks](https://github.com/gruntwork-io/terraform-aws-eks)
* [r:terraform-aws-elk](https://github.com/gruntwork-io/terraform-aws-elk)
* [r:terraform-aws-kafka](https://github.com/gruntwork-io/terraform-aws-kafka)
* [r:terraform-aws-lambda](https://github.com/gruntwork-io/terraform-aws-lambda)
* [r:terraform-aws-load-balancer](https://github.com/gruntwork-io/terraform-aws-load-balancer)
* [r:terraform-aws-messaging](https://github.com/gruntwork-io/terraform-aws-messaging)
* [r:terraform-aws-mongodb](https://github.com/gruntwork-io/terraform-aws-mongodb)
* [r:terraform-aws-monitoring](https://github.com/gruntwork-io/terraform-aws-monitoring)
* [r:terraform-aws-openvpn](https://github.com/gruntwork-io/terraform-aws-openvpn)
* [r:terraform-aws-sam](https://github.com/gruntwork-io/terraform-aws-sam)
* [r:terraform-aws-security](https://github.com/gruntwork-io/terraform-aws-security)
* [r:terraform-aws-server](https://github.com/gruntwork-io/terraform-aws-server)
* [r:terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets)
* [r:terraform-aws-zookeeper](https://github.com/gruntwork-io/terraform-aws-zookeeper)
* [r:terragrunt](https://github.com/gruntwork-io/terragrunt)
* [r:terratest](https://github.com/gruntwork-io/terratest)
* [r:repo-copier](https://github.com/gruntwork-io/repo-copier)


[Community Guidelines](https://github.com/gruntwork-io/knowledge-base/discussions/1)
[Getting help](https://gruntwork.io/support/)
[Gruntwork documentation](https://www.gruntwork.io)

**Comment by Etiene** [comment]
Hello, @preeti3012. The current alarm modules at `terraform-aws-monitoring` don't seem to support cross-account alarms yet, but let me double check and get back to you! (I believe one issue might be that it needs to define a `metric_query` instead of `metric_name`. More info [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_metric_alarm).)

**Comment by Etiene** [comment, marked as answer]
Hi, @preeti3012. Just confirming that the current modules in [`terraform-aws-monitoring`](https://github.com/gruntwork-io/terraform-aws-monitoring) do not support cross-account alarms, so this is a feature request. I created an item in our backlog to reflect it. You can follow, comment and vote here: [Cross-account alarms](https://github.com/gruntwork-io/terraform-aws-monitoring/issues/224). (PRs are also welcome!)
